### DISCUSSION TITLE
Transport Rework

### DISCUSSION THREAD

**Discussion by jazzysoggy** [OpenRCT2/OpenRCT2#17487] [Feature requests]
In the current version of OpenRCT2, transit is no different than any other ride, meaning that there is no purpose to using transit outside of moving guests back to the entrance from further corners of the map, which even this is finicky and requires you to only put exits at the entrance station. 

The current implementation also makes it difficult for setting up separated secluded areas of the park as the guests will only arrive there by chance and will get angry until they find the need to ride transit back.

I'm proposing a change where guests factor in transit as a part of their path-finding, and see whether transit is a viable option.

To factor in the difference in time for both options would be done via a different distance multiplier for each form of transit/walking.

Cost could be factored in by distance per dollar or by another multiplier. 

This way, transit is functional for something outside of just being another low intensity ride.

**Comment by IntelOrca** [comment]
> In the current version of OpenRCT2, transit is no different than any other ride

This is not true, guests will ride transport rides even if they are "leaving the park".

When we implement a new path finding system, guests will take rides into account.

**Reply by jjpb44** [reply]
What are the main objectives of a new route finding system? I think many suggested algorithms could work fine, but with thousands of guests it can increase CPU usage a lot.
